### PR TITLE
 Proposal: build iot system configuration isolation on nodepool(openyurtio#1597)

### PR DESCRIPTION
**Pull request by WoShiZhangmingyu**
#### Special notes for your reviewer:

@LavenderQAQ

### CODE REVIEW

File: docs/proposals/20240819-build-iot-system-configuration-isolation-on-nodepool.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+The platformadmin-controller, integrated within yurt-manager, is responsible for parsing the PlatformAdmin CR into the corresponding configmap, service, and yurtappset, thereby realizing the deployment of the edge device management platform. 
+
+Users have the capability to customize a PlatformAdminFramework that is initialized with a standard configuration, followed by the creation of a PlatformAdmin. After this step, the platformadmin-controller will automatically initiate the reconciliation process to handle services and YurtAppSets.
+
+However, a notable drawback of this process is that, even when multiple nodepools share identical configurations, it necessitates the creation of multiple PlatformAdmins.  This redundancy can lead to unnecessary administrative overhead and complexity.
+
+![platformadmin-old-frame](../img/20240819-build-iot-system-configuration-isolation-on-nodepool/platformadmin-old-frame.png)
```
Comment (rambohe-ch) [inline docs/proposals/20240819-build-iot-system-configuration-isolation-on-nodepool.md:57]: Auto-Controller --> Auto-Collector

Comment (WoShiZhangmingyu) [reply docs/proposals/20240819-build-iot-system-configuration-isolation-on-nodepool.md:57]: Now, the image content has been updated.

File: docs/proposals/20240819-build-iot-system-configuration-isolation-on-nodepool.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+Platformadmin needs to provide deployment for multiple nodepools, so the original **poolName** has been changed to the **nodePools** , as follows:
+~~~ 
+nodepools:
+    items:
+        type: string
+    type: array
+~~~
+A simple example:
+~~~
+apiVersion: iot.openyurt.io/v1alpha2
```
Comment (LavenderQAQ) [inline docs/proposals/20240819-build-iot-system-configuration-isolation-on-nodepool.md:87]: Consider upgrading iot api to v1beta1.

File: docs/proposals/20240819-build-iot-system-configuration-isolation-on-nodepool.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+}
+yas.Spec.Workload.WorkloadTweaks = []appsv1beta1.WorkloadTweak{
+	{
+		Pools: yas.Spec.Pools,
+		Tweaks: appsv1beta1.Tweaks{
+			Replicas: pointer.Int32(1),
+		},
+	},
+}
+~~~
```
Comment (LavenderQAQ) [inline docs/proposals/20240819-build-iot-system-configuration-isolation-on-nodepool.md:141]: Need to write how to upgrade to v1beta1, conversion webhook how to convert.

**Review by LavenderQAQ — CHANGES_REQUESTED** [review]

**Review by LavenderQAQ — APPROVED** [review]
/lgtm

**Pull request merged**